### ISSUE TITLE
Tests pick up configuration file in default location

### ISSUE THREAD

**Issue by geimer** [hpcugent/easybuild-framework#1467]
While running the framework test suite, I experienced failing tests since the configuration file `$HOME/.config/easybuild/config.cfg` was picked up and evaluated.

**Comment by wpoely86** [comment]
Which tests? Can you give use the debug output of the tests?

**Comment by geimer** [comment]
In my particular case, setting `sourcepath` in `$HOME/.config/easybuild/config.cfg` led to three failures:
 * test.framework.config
```
======================================================================
FAIL: test_default_config (__main__.EasyBuildConfigTest)
Test default configuration.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/geimer/Projects/EasyBuild/easybuild-framework/test/framework/config.py", line 119, in test_default_config
    self.assertEqual(config_options['sourcepath'], os.path.join(eb_homedir, 'sources'))
AssertionError: '/home/geimer/Projects/EasyBuild/Archives/Sources' != '/home/geimer/.local/easybuild/sources'

======================================================================
FAIL: test_generaloption_config_file (__main__.EasyBuildConfigTest)
Test use of new-style configuration file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/geimer/Projects/EasyBuild/easybuild-framework/test/framework/config.py", line 277, in test_generaloption_config_file
    self.assertEqual(source_paths(), [os.path.join(os.getenv('HOME'), '.local', 'easybuild', 'sources')])  # default
AssertionError: Lists differ: ['/home/geimer/Projects/EasyBu... != ['/home/geimer/.local/easybuil...

First differing element 0:
/home/geimer/Projects/EasyBuild/Archives/Sources
/home/geimer/.local/easybuild/sources

- ['/home/geimer/Projects/EasyBuild/Archives/Sources']
+ ['/home/geimer/.local/easybuild/sources']

----------------------------------------------------------------------
```
 * test.framework.options
```
FAIL: test_generate_cmd_line (__main__.CommandLineOptionsTest)
Test for generate_cmd_line.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/geimer/Projects/EasyBuild/easybuild-framework/test/framework/options.py", line 1674, in test_generate_cmd_line
    self.assertEqual(ebopts.generate_cmd_line(), [])
AssertionError: Lists differ: ['--sourcepath=/home/geimer/Pr... != []

First list contains 1 additional elements.
First extra element 0:
--sourcepath=/home/geimer/Projects/EasyBuild/Archives/Sources

- ['--sourcepath=/home/geimer/Projects/EasyBuild/Archives/Sources']
+ []

----------------------------------------------------------------------
```
@boegel and I discussed this issue already at the Hackathon last week and came to the conclusion that this should be handled in the framework, as other settings may (and in fact do) affect other tests.

**Comment by boegel** [comment]
same issue being reported in #1478

**Comment by boegel** [comment]
fixed in #1513
